skimage/restoration/_denoise_cy.pyx

```diff
@@ -119,7 +119,7 @@ def _denoise_tv_bregman(np_floats[:, :, ::1] image, np_floats weight,
         np_floats ux, uy, uprev, unew, bxx, byy, dxx, dyy, s, tx, ty
         int i = 0
         np_floats lam = 2 * weight
-        np_floats rmse = DBL_MAX
+        double rmse = DBL_MAX
```

Comment:
why was this line changed, the whole point of this was to keep float32 as float32 as long as possible. wasa it causing errors?

Reply:
I *presume* that setting a float32 to DBL_MAX does not do good things. Perhaps I there is a FLT_MAX?

Reply:
ah i see. I didn't know what DBL meant, there is a way to switch code paths between dtypes for a fused type. I guess we would need to do that here.

Reply:
We could just set it to FLT_MAX regardless of dtype. If your initial RMSE exceeds that, you're in deep trouble :joy:

Anyway, the code works, it's just probably a tiny bit slower because of float32-float64 comparisons in the float32 input case.

Reply:
yeah, this "optimization" would be a 0.16 branch only IMO.